packages/apps/dashboard/server/src/modules/networks/networks.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { ChainId, NETWORKS, StatisticsClient } from '@human-protocol/sdk';
+import { Cache, CACHE_MANAGER } from '@nestjs/cache-manager';
+
+import {
+  EnvironmentConfigService,
+  MINIMUM_ESCROWS_COUNT,
+  MINIMUM_HMT_TRANSFERS,
+} from '../../common/config/env-config.service';
+import { OPERATING_NETWORKS_CACHE_KEY } from '../../common/config/redis-config.service';
+import { MAINNET_CHAIN_IDS } from '../../common/utils/constants';
+
+@Injectable()
+export class NetworksService {
+  private readonly logger = new Logger(NetworksService.name);
+  constructor(
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
+    private readonly envConfigService: EnvironmentConfigService,
+  ) {}
+
+  public async getOperatingNetworks(): Promise<ChainId[]> {
+    const cachedNetworks = await this.cacheManager.get<ChainId[]>(
+      OPERATING_NETWORKS_CACHE_KEY,
+    );
+
+    if (cachedNetworks) {
+      return cachedNetworks;
+    }
+
+    const currentMonth = new Date();
+    const oneMonthAgo = new Date(currentMonth);
+    oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+
+    const filterDate = new Date(currentMonth);
+    filterDate.setMonth(
+      filterDate.getMonth() - this.envConfigService.networkUsageFilterMonths,
+    );
+
+    const availableNetworks = [];
+
+    for (const chainId of Object.values(MAINNET_CHAIN_IDS)) {
+      const networkConfig = NETWORKS[chainId];
+      if (!networkConfig) continue;
+
+      const statisticsClient = new StatisticsClient(networkConfig);
+      try {
+        const [hmtData, escrowStats] = await Promise.all([
+          statisticsClient.getHMTDailyData({ from: filterDate }),
```

Comment:
You have to keep
```
from: new Date(Math.floor(filterDate.getTime() / 1000) * 1000)
```
here, because under the hood for querying subgraph it uses seconds (and [does implicit conversion](https://github.com/humanprotocol/human-protocol/blob/32eda6b952c6cc70c4561cca32a8a17736768ba3/packages/sdk/typescript/human-protocol-sdk/src/statistics.ts#L509)), so if you don't `Math.floor` here - it fails with error like
```
Error processing network 56: An error occurred while interacting with Ethereum: Invalid value provided for argument `from`: Float(1731492900.033)
```

It's very confusing that we have such conversion here, but it's better to keep it as it was and fix SDK as separate PR

For ref: https://github.com/humanprotocol/human-protocol/issues/2912